rllib/evaluation/sampler.py

```diff
@@ -1007,13 +1025,31 @@ def _process_observations(
         # (no step taken yet, just a single starting observation logged).
         # We need to skip this callback in this case.
         if not episode.is_faulty and episode.length > 0:
-            callbacks.on_episode_step(
-                worker=worker,
-                base_env=base_env,
-                policies=worker.policy_map,
-                episode=episode,
-                env_index=env_id,
-            )
+            if worker.policy_config["in_evaluation"]:
```

Comment:
I want to make a high level suggestion.
I don't think you really need on_evaluate_step() callback, because you have access to the worker in on_episode_step(), which you can use to decide whether this is an evaluation episode or a normal rollout episode.
it's actually a bit weird to only call on_evaluate_step() for eval episodes. because an episode step is an episode step, if someone wants to modify an episode step in any way, the modification needs to be applied to both rollout and eval episodes.
so my suggestion is to only add the on_evaluate_start() and on_evaluate_end() callbacks, and leverage the existing on_episode_step() to do what you want.
wdyt?

Reply:
I agree that the on_evaluate_step() is not really needed, but I included it to have the API look the same as the episode callbacks. However, I think that it may be a nicer way to access the information on whether the episode is evaluation or normal this way. 

As for the episode modification, I am not fully familiar with ray internals yet, but would there be no use case for modifying just the eval episode?

Either way, I do not use the on_evaluate_step() myself, and if you think that it is better to be removed, please let me know, and I can do so!

Reply:
let me suggest we do not add the on_evaluate_step() for now, and that will make this PR much simpler too since you can undo all those if else changes.
and we can definitely revisit later in case there is a need.
can you help remove some of the changes here?

Reply:
Okay, I have removed the on_evaluate_step(), and the corresponding functions in other files.